Manager fails after starting a repair: `panic: runtime error: invalid memory address or nil pointer dereference`

After starting a standard repair (i.e with no additional arguments at all), the manager crashes. The error as printed by the log:
```
{"L":"INFO","T":"2021-12-13T11:01:47.391+0200","N":"repair","M":"Detected small tables","tables":["system_distributed.view_build_status","system_distributed.service_levels","system_distributed.cdc_streams_descriptions_v2","system_distributed.cdc_generation_timestamps","system_distributed_everywhere.cdc_generation_descriptions_v2","system_traces.sessions","system_traces.sessions_time_idx","system_traces.node_slow_log_time_idx","system_traces.events","system_traces.node_slow_log","ks.cf","scylla_manager.repair_run","scylla_manager.scheduler_task","scylla_manager.repair_run_progress","scylla_manager.backup_run","scylla_manager.cluster","scylla_manager.secrets","scylla_manager.scheduler_task_run","scylla_manager.validate_backup_run_progress","scylla_manager.backup_run_progress","scylla_manager.gocqlx_migrate","scylla_manager.repair_run_state","scylla_manager.drawer"],"threshold":1073741824,"_trace_id":"_8nZjV-sSs-gQS0Y4z97Jg"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12ae017]

goroutine 740 [running]:
github.com/scylladb/scylla-manager/pkg/scyllaclient.(*Client).TotalMemory(0xc000183540, 0x2d4fa68, 0xc000489140, 0xc0006f81e0, 0xa, 0x0, 0xc000ce6b40, 0xc000822e08)
        github.com/scylladb/scylla-manager/pkg/scyllaclient/client_scylla.go:771 +0x117
github.com/scylladb/scylla-manager/pkg/service/repair.(*Service).hostRangeLimits.func1(0x1, 0xc000849b60, 0x0)
        github.com/scylladb/scylla-manager/pkg/service/repair/service.go:509 +0xd0
github.com/scylladb/scylla-manager/pkg/util/parallel.Run.func1(0xc0009e41f0, 0x3, 0xc0009e41f8, 0xc000c92060, 0xc000de2000)
        github.com/scylladb/scylla-manager/pkg/util/parallel/parallel.go:69 +0x7f
created by github.com/scylladb/scylla-manager/pkg/util/parallel.Run
        github.com/scylladb/scylla-manager/pkg/util/parallel/parallel.go:54 +0x132
```
Manager build: http://downloads.scylladb.com/manager/relocatable/unstable/branch-2.6/2021-12-12T21:03:50Z/scylla-manager_2.6.rc0-0.20211212.dfb1d2b628d.linux_x86_64.tar.gz

--- a/pkg/scyllaclient/client_scylla.go
+++ b/pkg/scyllaclient/client_scylla.go
@@ -760,7 +760,12 @@ func (c *Client) TotalMemory(ctx context.Context, host string) (int64, error) {
 
 	var totalMemory int64
 	for _, m := range metrics[metricName].Metric {
-		totalMemory += int64(*m.Counter.Value)
+		switch {
+		case m.Counter != nil:
+			totalMemory += int64(*m.Counter.Value)
+		case m.Gauge != nil:
+			totalMemory += int64(*m.Gauge.Value)
+		}
 	}
 
 	return totalMemory, nil

--- a/pkg/scyllaclient/client_scylla_test.go
+++ b/pkg/scyllaclient/client_scylla_test.go
@@ -204,6 +204,21 @@ func TestClientShardCount(t *testing.T) {
 	}
 }
 
+func TestClientTotalMemory(t *testing.T) {
+	t.Parallel()
+
+	client, closeServer := scyllaclienttest.NewFakeScyllaServer(t, "testdata/scylla_metrics/metrics")
+	defer closeServer()
+
+	v, err := client.TotalMemory(context.Background(), scyllaclienttest.TestHost)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if v != 31138512896 {
+		t.Fatal(v)
+	}
+}
+
 func TestClientDescribeRing(t *testing.T) {
 	t.Parallel()
 

